# **1. Health Expectation**

In [1]:
import seaborn as sns
from scipy.stats import pearsonr
from scipy.stats import ttest_ind
from scipy.stats import ttest_1samp

In [2]:
# load dataset
df = sns.load_dataset('healthexp')
df.head()

Year        Country  Spending_USD  Life_Expectancy
0  1970        Germany       252.311             70.6
1  1970         France       192.143             72.2
2  1970  Great Britain       123.993             71.9
3  1970          Japan       150.437             72.0
4  1970            USA       326.961             70.9

1. Corrélation de Person entre les deux variables continues Spending_USD et Life_Expectancy

In [3]:
df_usa = df[(df['Country'] == 'USA') & (df['Year'] >= 2000) & (df['Year'] <= 2020)]
pearsonr(df_usa['Spending_USD'], df_usa['Life_Expectancy'])

PearsonRResult(statistic=0.6210674528998252, pvalue=0.0026581544805799094)

- La corrélation de Pearson est de 62.1%
- Si on part de l'hypothèse nulle qu'en réalité ces deux variables ne sont pas corrélées, la probabilté pour observer une corrélation de 62.1% est 
de 0.2%, ce qui est en dessous du seuil aplha. On peut donc dire que notre hypothèse H0, n'est pas correcte. 


2. Dans les vidéos précédentes, il semblait également que l'Allemagne et la Grande-Bretagne avaient la même durée de vie moyenne de 1970 à 2020.



In [4]:
# df_germany = df[(df['Country'] == 'Germany') & (df['Year'] >= 1970) & (df['Year'] <= 2020)]
# df_great_britain = df[(df['Country'] == 'Great Britain') & (df['Year'] >= 1970) & (df['Year'] <= 2020)]

df_germany_test = df.query('Country == "Germany" & Year >= 1970 & Year <= 2020')
df_great_britain_test = df.query('Country == "Great Britain" & Year >= 1970 & Year <= 2020')

ttest_ind(df_germany_test['Spending_USD'], df_great_britain_test['Spending_USD'])

TtestResult(statistic=1.8067367331266013, pvalue=0.07410847965729157, df=91.0)

- la p-value est de 7.4%, ce qui est au dessus de 2%, nous n'avons donc pas assez d'évidence pour rejeter H0. 


3. Un politicien américain affirme que depuis 1970, la durée de vie aux États-Unis augmente en moyenne de 0,3 an. Testez cette affirmation et donnez votre conclusion avec un niveau de confiance de 98 % (alpha = 0,02). (Vous pouvez utiliser la fonction `.diff()` de pandas pour calculer la différence d'une ligne à l'autre dans votre DataFrame)

- Ho = La moyenne d'augmentation annuelle de l'âge aux usa de 1970 à nos jours est de 0,3 an. 

In [14]:
df_usa = df.query('Country == "USA"')
df_usa['Life_Expectancy'] = df_usa['Life_Expectancy'].diff()


C:\Users\Wolf\AppData\Local\Temp\ipykernel_6572\1291104361.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_usa['Life_Expectancy'] = df_usa['Life_Expectancy'].diff()


- après observation, celle-ci est de 0.121 an

In [15]:
ttest_1samp(df_usa['Life_Expectancy'].dropna(), popmean=0.3)

TtestResult(statistic=-3.7212105613188773, pvalue=0.0005112013249193949, df=49)

On peut rejeter H0 parce que la p-value est tr-s très faible...